pyaerocom/io/cams2_83/obs.py

```diff
@@ -71,4 +74,28 @@ def read_csv(
     if (df.conc <= 0).any():
         logger.warning("found negative obs")
         df = df[df.conc > 0]
+    metadata_file_path = Path(path).parent.parent / DEFAULT_METADATA_NAME
+    if metadata_file_path.is_file():
+        try:
+            df_metadata = read_metadata(metadata_file_path)
+        except ParserError as e:
+            logger.warning(f"Invalid metadata file {path}, {e}")
+            df_metadata = pd.DataFrame()
+        if not df_metadata.empty:
+            df = df.merge(df_metadata, on="station", how="left")
+            return df["station lat lon alt time poll conc station_type".split()]
+        else:
+            logger.warning("Empty metadata")
+    else:
+        logger.warning(f"Metadata file {metadata_file_path} does not exist")
```

Comment:
The logic with/without metadata is a bit hard to follow, maybe refactoring with a guarding clauses will make it easier to follow

``` python
    df = df["station lat lon alt time poll conc".split()]

    metadata_file_path = Path(path).parent.parent / DEFAULT_METADATA_NAME
    if not metadata_file_path.is_file():
        logger.warning(f"Metadata file {metadata_file_path} does not exist")
        return df

    try:
        df_metadata = read_metadata(metadata_file_path)
    except ParserError as e:
        logger.warning(f"Invalid metadata file {path}, {e}")
        return df
    
    if df_metadata.empty:
       logger.warning("Empty metadata")
       return df
            
    return df.merge(df_metadata, on="station", how="left")
```

Reply:
yeah sure, thanks for the suggestion, this reads better